### PR TITLE
Add docstrings to classify_tabs.py

### PR DESCRIPTION
**Pull request by rusheb**
I added docstrings to classify_tabs.py as I was reading through the code and understanding it. I also did some small refactors (mostly extracting variables) which I think make it more readable. 

I made one functional change: `generate_continuation` will now print the continuation up to and *including* the stop token. I think this makes more sense because it tells the user that the stop token was reached, but I can change it back if you disagree. 

There is one TODO in the docstrings, because I didn't understand why we don't return the last logit in `get_logits_and_tokens`. It seems like we do this to make it easier to get the logits for the last token in `test_generation`. However this would break down if we hit `max_length` instead of `stop_token`. 

As an aside, I think we should move the test functions into a test package so that they don't clutter up the main implementation. Let me know if you'd like me to do this. Maybe it's premature. 

I'm not sure whether we should merge this PR as I don't know if it adds value, but I thought I would raise the PR and let you decide.

### CODE REVIEW

**Comment by mivanit** [comment]
I generally tend to follow a docstring format similar to numpydoc (markdown-formatted is nice), So I apologize if my existing/future docstrings don't always conform. definitely not a big deal for a project of this scale
